documentation/milestones/individual_milestones/developer_tools.md

```diff
@@ -0,0 +1,221 @@
+  - [ ] Code which fails any of the unit or integration tests will be blocked from merging into `main`
+  - [ ] A report on the percentage of code covered by tests is available after every test run
+  - [ ] Code whose test coverage falls below a certain threshold will be blocked from merging into `main`
+- [ ] **Code quality checks** are live and meet the following conditions:
+  - [ ] All checks are run on every push to the remote GitHub repository
+  - [ ] The most important checks are run on every local commit
+  - [ ] Code which fails any of these checks will be blocked from merging into `main`
+- [ ] **Security checks** are live and meet the following conditions:
+  - [ ] At least 1 (test) security vulnerability report has been submitted
+  - [ ] Maintainers are notified within 1 hour of a vulnerability being reported within the grants API codebase
+  - [ ] Maintainers are notified within 72 hours of a vulnerability being reported on an upstream dependency
+  - [ ] Security checks are running on every push to the remote GitHub repository
+  - [ ] Code which fails any of these security checks is blocked from merging into `main`
+- [ ] **Dependency management** is live and meets the following conditions:
+  - [ ] At least 2 upstream dependencies have been added to the project
+  - [ ] Compatible versions of upstream dependencies are automatically detected when a new dependency is added
+  - [ ] Maintainers are notified when a new minor or major version of an upstream dependency is available
+- [ ] **Config & secrets management** is live and meets the following conditions:
+  - [ ] At least 1 configuration variable has been added to the project
+  - [ ] This config variable has different values when it is running in different environments (e.g. dev, staging, and prod)
+  - [ ] At least 2 internal developers can access a shared value of this variable locally
+  - [ ] Developers can change the value of this config variable in their local environment without changing it for other developers
+  - [ ] An integration test has been created to validate that this variable is injected at runtime
+
+### Proposed metrics for measuring goals/value/definition of done
+
+- Test framework metrics
+  - Percentage of code covered by unit tests
+  - Percentage of code covered by integration tests
+  - Total runtime of all unit tests
+  - Total runtime of all integration tests
+- Code quality checks metric: Total runtime of all code quality checks
+- Security checks metrics
+  - Total runtime of all security checks
+  - Number of security vulnerabilities not caught until production
+  - Number of days it takes to resolve a security vulnerability reported in the grants API project
+  - Number of days it takes to update an upstream dependency once a version which resolves a CVE has been released
+- Dependency management metrics
+  - Number of days it takes to update an upstream dependency once a new minor or major version has been released
```

Comment:
Hmm. Some dev teams prefer to leave dependencies untouched until you need new features (or there's a security vulnerability), given that updates occasionally break things. So it may not always be needed for us to update when new versions have been released.

Reply:
@lucasmbrown-usds This is worth discussing further! 

@daphnegold was proposing the use of Renovate in the [FE code quality tools](https://github.com/HHS/grants-equity/pull/190), which automates the creation of PRs to adopt dependency updates, but you make a good point that a fair amount of time can be sunk into simply making updates when they're not needed.

On past projects I've found regularly updating dependencies to be helpful because otherwise teams can wind up scrambling to make pretty significant updates once a CVE is identified, or they delay updates for years because they become so large and painful to make -- e.g. in my previous project the team was still using Airflow 1.x and had tried and failed to upgrade to Airflow 2.x  a few times because they were so far behind on updates.

That being said I like the idea of striking this as a core metric and took it out of the list!

Reply:
If you and the dev team prefer to keep everything updated, then I'm fine adding this metric back in!